homework/list.js

```diff
@@ -1,79 +1,149 @@
 'use strict';
 
-(function() {
+/**
+ * @typedef {Object} Note
+ * @property {String} title
+ * @property {String} passages
+ */
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  /**
+   * Create a instance of ListManager
+   * @constructor
+   * @this {ListManager}
+   */
+  var ListManager = function(){
+      this._wrapper = document.querySelector('#note-list-wrapper');
+      this._listNoteContent = [];
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
+  /**
+   * Fetch the notes from remote server, draw them in the notes wrapper,
+   * and bind click event to dispatch note-open event
+   *
+   * @this {ListManager}
+   */
+  ListManager.prototype.start = function(){
+      this.fetchList().then((function(data) {
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this)).catch((function () {
+        // Deal with fetch error
+      }).bind(this));
+      window.addEventListener('click', (function(event) {
+        this.onNoteOpen(event);
+      }).bind(this));
+  };
+
+
+  /**
+   * Dispatch the note-open event and prepare the param for the dispatched event
+   *
+   * @this {ListManager}
+   */
+  ListManager.prototype.onNoteOpen = function(){
+      if (event.target.classList.contains('note-title')) {
+        var id = event.target.dataset.noteId;
+        var content = this._listNoteContent[id];
+        window.dispatchEvent(new CustomEvent('note-open',
+          { detail: content }));
+      };
+  },
+
+
+  /**
+   * Preload the first note at the first time this instance start
+   *
+   * @this {ListManager}
+   */
+  ListManager.prototype.preloadFirstNote = function() {
+    if (this._listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped with one conditional statement with only one branch, the `if...else` should be put in the caller, rather than the callee.